### PR TITLE
Fix graph break in torch.compile when using FA2 with attention_mask=None and batch size > 1

### PR DESCRIPTION
**Pull request by efsotr**
# What does this PR do?

Fixes partial https://github.com/huggingface/transformers/issues/37199#issue-2965516901

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@ArthurZucker

### CODE REVIEW

**Review by ArthurZucker — COMMENTED** [review]
Thanks, can we add a test ! 🤗

File: tests/test_modeling_common.py [hunk 1]
```diff
@@ -4021,6 +4021,43 @@ def test_flash_attn_2_fp32_ln(self):
...
+            "input_ids": torch.randint(low=1, high=model.config.vocab_size, size=(2, 10), device=torch_device),
+            "labels": torch.randint(low=1, high=model.config.vocab_size, size=(2, 10), device=torch_device),
+        }
+
+        model = torch.compile(model, fullgraph=True)
+        # forward compilation
+        set_seed(42)
+        loss = model(**inputs).loss
+        # backward compilation
+        loss.backward()
```
Comment (ArthurZucker) [inline tests/test_modeling_common.py:4059]: can we assert loss its not nan?

**Review by ArthurZucker — APPROVED** [review]

File: tests/test_modeling_common.py (hunk 1 above)
Comment (efsotr) [reply tests/test_modeling_common.py:4059]: Logically, this test is only checking whether there is a graph break in torch.compile.

Comment (efsotr) [reply tests/test_modeling_common.py:4063]: assert added

**Comment by efsotr** [comment]
due to #37092 , the added test can pass when export DISABLE_KERNEL_MAPPING=1

**Comment by llllvvuu** [comment]
Is this ready to merge? @ArthurZucker

**Comment by efsotr** [comment]
ready to merge! @ArthurZucker

**Comment by ArthurZucker** [comment]
Merging sorry!
